cpp/references/data_structures/list_digraph.cpp

```diff
@@ -1 +1,288 @@
-;
\ No newline at end of file
+#include "list_digraph.hpp"
+
+using namespace godot;
+
+
+/* ########################### VERTEX ########################### */
+
+typedef ListDigraph::Vertex Vertex;
+
+/* ##### GDNATIVE API ##### */
+
+Vertex *Vertex::_new(lemon::ListDigraph::Node p_vertex, ListDigraph *p_graph) {
+    Vertex* vertex = _new();
+    vertex->_vertex = p_vertex;
+    vertex->_graph = p_graph;
+    return vertex;
+}
+
+lemon::ListDigraph::Node Vertex::get_vertex() const {
+    return _vertex;
+}
+
+/* ##### GDSCRIPT API ##### */
+
+void Vertex::_init() {
+	
+}
+
+Vertex *Vertex::split(const bool p_connect) {
+	return Vertex::_new(_graph->_graph.split(_vertex, p_connect), _graph);
+}
+
+int Vertex::get_id() const {
+	return _graph->_graph.id(_vertex);
+}
+
+bool Vertex::is_valid() const {
+    return _graph->_graph.valid(_vertex);
+}
+
+Variant Vertex::get_data() const {
+    return (*_graph->_vertex_data)[_vertex];
+}
+
+void Vertex::set_data(Variant p_value) {
+    (*_graph->_vertex_data)[_vertex] = p_value;
+}
+
+void Vertex::_register_methods() {
+	register_method("get_id", &Vertex::get_id);
+	register_method("is_valid", &Vertex::is_valid);
+	register_method("get_data", &Vertex::get_data);
+	register_method("set_data", &Vertex::set_data);
+}
+
+/* ########################### ARC ########################### */
+
+typedef ListDigraph::Arc Arc;
+
+/* ##### GDNATIVE API ##### */
+
+Arc *Arc::_new(lemon::ListDigraph::Arc p_arc, ListDigraph *p_graph) {
+    Arc* arc = _new();
+    arc->_arc = p_arc;
+    arc->_graph = p_graph;
+    return arc;
+}
+
+lemon::ListDigraph::Arc Arc::get_arc() const {
+    return _arc;
+}
+
+/* ##### GDSCRIPT API ##### */
+
+void Arc::_init() {
+
+}
+
+void Arc::reverse() {
+	_graph->_graph.reverseArc(_arc);
+}
+
+Vertex *Arc::split() {
+	return Vertex::_new(_graph->_graph.split(_arc), _graph);
+}
+
+int Arc::get_id() const {
+	return _graph->_graph.id(_arc);
+}
+
+bool Arc::is_valid() const {
+    return _graph->_graph.valid(_arc);
+}
+
+Variant Arc::get_data() const {
+    return (*_graph->_arc_data)[_arc];
+}
+
+void Arc::set_data(Variant p_value) {
+	(*_graph->_arc_data)[_arc] = p_value;
+}
+
+void Arc::set_source(const Vertex *p_vertex) {
+	_graph->_graph.changeSource(_arc, p_vertex->get_vertex());
+}
+
+Vertex *Arc::get_source() {
+	return Vertex::_new(_graph->_graph.source(_arc), _graph);
+}
+
+void Arc::set_target(const Vertex *p_vertex) {
+	_graph->_graph.changeTarget(_arc, p_vertex->get_vertex());
+}
+
+Vertex *Arc::get_target() {
+	return Vertex::_new(_graph->_graph.target(_arc), _graph);
+}
+
+void Arc::_register_methods() {
```

Comment:
Not only data, like with the Vertex, but we can register `set/get_source`, `set/get_target`, and then register them each as properties.